scripts/start_registry.sh

```diff
@@ -0,0 +1,12 @@
+#!/usr/bin/env bash
+set -o errexit
+
+# create registry container unless it already exists
+reg_name='kind-registry'
+reg_port='5000'
+running="$(docker inspect -f '{{.State.Running}}' "${reg_name}" 2>/dev/null || true)"
+if [ "${running}" != 'true' ]; then
+  docker run \
```

Comment:
is it possible to spin up this registry from within the cluster instead?

Reply:
This is a major rabbit hole, but here we go...

Testing with either SSL or non-SSL registries has its own set of problems due to the way docker daemon needs to be configured. Note that the operator-registry tests skip over this whole problem for docker because of the issues involved. Note that operator-registry has its own in-process docker registry that can start up and run and it can create a CA cert that it can hand to the containerd tooling to make things work with SSL. This won't work for docker tooling because the CA cert has to be available to the docker daemon. If you want to configure a per-registry SSL configuration there is a mechanism for this, but you have to configure it and then restart the docker daemon (see 
https://docs.docker.com/engine/security/certificates/). If you want to setup insecure registries via the `insecure-registry` configuration, you can do that, but again it requires the docker daemon to be restarted (see https://docs.docker.com/registry/insecure/). 

Bottom line for SSL is if you can get the system to trust your CA, then things go easier. Note that using a local docker registry from **within a container** works around a bunch of different problems relating to networking and getting SSL to work. The docker network can allow you to bridge networks between containers (which is what's happening for the kind server in this PR). I've also seen that it makes insecure registries work properly with docker push/pull (I presume this is working because of the out of box host networking configuration).

Note that we put a variation of this script, which is far more capable, into https://github.com/operator-framework/operator-registry/blob/master/scripts/start_registry.sh where we spin up a registry. Note that it will correctly handle setting up CA certs in a cross platform way since it is using [mkcert](https://github.com/FiloSottile/mkcert) under the hood to do this "right". 

One idea I just had would be to make use of this more capable shell script but execute it when the testing actually starts, which would remove the extra step that a user would have to do to execute the tests. You can also just change the make file call the script too, which accomplishes the same thing.